go/enclave/components/gas_estimate.go

```diff
@@ -27,37 +27,68 @@ var AdjustPublishingGas = gethcommon.Big2
 
 type GasEstimator struct {
 	storage   storage.Storage
+	registry  BatchRegistry
 	chain     TENChain
 	logger    gethlog.Logger
 	gasOracle gas.Oracle
 	gasPricer *GasPricer
 }
 
-func NewGasEstimator(storage storage.Storage, chain TENChain, gasOracle gas.Oracle, gasPricer *GasPricer, logger gethlog.Logger) *GasEstimator {
+func NewGasEstimator(storage storage.Storage, registry BatchRegistry, chain TENChain, gasOracle gas.Oracle, gasPricer *GasPricer, logger gethlog.Logger) *GasEstimator {
 	if gasPricer == nil {
 		logger.Crit("gasPricer cannot be nil - this indicates a critical initialization failure")
 	}
 	return &GasEstimator{
 		storage:   storage,
+		registry:  registry,
 		chain:     chain,
 		gasOracle: gasOracle,
 		gasPricer: gasPricer,
 		logger:    logger,
 	}
 }
 
-func (ge *GasEstimator) EstimateTotalGas(ctx context.Context, args *gethapi.TransactionArgs, blockNumber *gethrpc.BlockNumber, batch *common.BatchHeader, globalGasCap uint64) (uint64, uint64, error, common.SystemError) {
+func (ge *GasEstimator) EstimateTotalGas(ctx context.Context, args *gethapi.TransactionArgs, blockNumber gethrpc.BlockNumber, globalGasCap uint64) (uint64, uint64, error, common.SystemError) {
+	var headBatch *common.BatchHeader
+	var estimationBatchHeight gethrpc.BlockNumber
+	var err error
+
+	// don't use the latest batch for gas estimation because it might influence the next batch (possible bug)
+	if blockNumber == gethrpc.LatestBlockNumber || blockNumber == gethrpc.PendingBlockNumber || blockNumber == gethrpc.FinalizedBlockNumber || blockNumber == gethrpc.SafeBlockNumber {
```

Comment:
this if is brutal. im guessing the comment is stale from the verkle trees cause we pull the latest now?

Reply:
I'll get rid of this, now that I reverted